external/web3-blockchain-service.js

```diff
@@ -26,88 +26,72 @@ class Web3BlockchainService {
     }
 
     async getGasStationPrice() {
-        const response = await axios.get(this.gasStationLink)
-            .catch((err) => {
-                this.logger.warn(err);
-                return undefined;
-            });
+        const response = await axios.get(this.gasStationLink).catch((err) => {
+            this.logger.warn(err);
+            return undefined;
+        });
         if (response) {
-            return response.data.standard * 1000000000;
+            return response.data.standard * 1e9;
         }
         return undefined;
     }
 
-    async createAssertionRecord(stateCommitHash, rootHash, issuer) {
-        const contractAddress = await this.getAssertionRegistryAddress();
-        const contractInstance = new this.web3.eth.Contract(DKGContract, contractAddress);
+    async executeContractMethod(contractInstance, method, args) {
+        let gasPrice = await this.getGasStationPrice();
+        gasPrice = Math.round(gasPrice);
```

Comment:
@kotlarmilos Do you remember which issues we had with gas station previously? If I am not wrong it was something regarding rounding the velue that is returned by gas station

Reply:
Rounding shouldn't be an issue since you multiply by 1e9 before doing it. Something that I added is estimation of gas limit, which I think was hardcoded